lib/Doctrine/DBAL/Platforms/DB2Platform.php

```diff
@@ -771,16 +771,26 @@ public function getTemporaryTableName($tableName)
      */
     protected function doModifyLimitQuery($query, $limit, $offset = null)
     {
-        if ($limit === null && $offset === null) {
-            return $query;
+        $where = array();
+
+        if ($offset > 0) {
+            $where[] = sprintf('db22.DC_ROWNUM >= %d', $offset + 1);
         }
 
-        $limit = (int) $limit;
-        $offset = (int) (($offset)?:0);
+        if ($limit !== null) {
+            $where[] = sprintf('db22.DC_ROWNUM <= %d', $offset + $limit);
+        }
+
+        if (count($where) === 0) {
```

Comment:
Sure. Just trying to understand the requirements, is `empty($array)` somehow better than `count($array)`? IMO, `empty()` is prone to typos like `empty($were)` which one may not notice w/o running tests.